src/context.c

```diff
@@ -367,3 +369,35 @@ xkb_context_set_user_data(struct xkb_context *ctx, void *user_data)
 {
     ctx->user_data = user_data;
 }
+
+XKB_EXPORT enum xkb_warning_flags
+xkb_context_get_warning_flags(struct xkb_context *ctx)
+{
+    if (ctx)
+        return ctx->warning_flags;
+    return -1;
+}
+
+XKB_EXPORT void
+xkb_context_set_warning_flags(struct xkb_context *ctx,
+                              enum xkb_warning_flags warning_flags)
+{
+    ctx->warning_flags = warning_flags;
+    ctx->error_flags &= ~warning_flags;
```

Comment:
tbh, I don't think this bit is a good idea - adding a warning flag shouldn't remove an error flag. It's fine for warning upgraded to error but downgrading is not good. Errors default to 0 anyway so it's easier to handle that case.